source/postprocess/point_values.cc

```diff
@@ -0,0 +1,270 @@
+    {
+      // evaluate the solution at all of our evaluation points
+      std::vector<Vector<double> >
+      current_point_values (evaluation_points.size(),
+                            Vector<double> (this->introspection().n_components));
+
+      for (unsigned int p=0; p<evaluation_points.size(); ++p)
+        {
+          // try to evaluate the solution at this point. in parallel, the point
+          // will be on only one processor's owned cells, so the others are
+          // going to throw an exception. make sure at least one processor
+          // finds the given point
+          bool point_found = false;
+          try
+            {
+              VectorTools::point_value(this->get_dof_handler(),
+                                       this->get_solution(),
+                                       evaluation_points[p],
+                                       current_point_values[p]);
+              point_found = true;
+            }
+          catch (const VectorTools::ExcPointNotAvailableHere &)
+            {
+              // ignore
+            }
+
+          // ensure that exactly one processor found things
+          Assert (Utilities::MPI::sum (point_found ? 1 : 0, this->get_mpi_communicator()) > 0,
+                  ExcMessage ("While trying to evaluate the values of the solution at "
+                              "evaluation point " + Utilities::int_to_string(p) +
+                              ", no processor reported that that point lies inside the "
+                              "set of cells it owns. Are you trying to evaluate the "
+                              "solution at a point that lies outside the domain?"));
+
+          // now exchange things. because we have exactly one processor that found
+          // the point, we can just add up that value plus all of the zero
+          // vectors from the other processors
+          //
+          // at the time of writing this (where we require deal.II 8.2) the
+          // Utilities::MPI::sum function can't sum Vector<double> arguments, so
+          // convert everything into a std::vector and back
+          std::vector<double> v (current_point_values[p].begin(),
+                                 current_point_values[p].end());
+          Utilities::MPI::sum (v, this->get_mpi_communicator(),
+                               v);
+          std::copy (v.begin(), v.end(), current_point_values[p].begin());
+        }
+
+      // finally push these point values all onto the list we keep
+      point_values.push_back (std::make_pair (this->get_time(),
+                                              current_point_values));
+
+      // now write all of the data to the file of choice. start with a pre-amble that
+      // explains the meaning of the various fields
+      const std::string filename = (this->get_output_directory() +
+                                    "point_values.txt");
+      std::ofstream f (filename.c_str());
```

Comment:
Do you want to add an assertion that the file creation did work? In other parts of the code we usually have one.